hub/core/storage/provider_mapper.py

```diff
@@ -0,0 +1,147 @@
+from collections.abc import MutableMapping
+from typing import Optional
+from hub.util.assert_byte_indexes import assert_byte_indexes
+from hub.constants import BYTE_PADDING
+from hub.core.storage.provider import Provider
+
+
+class ProviderMapper(Provider):
```

Comment:
Can we rename this to `MappedProvider` so that it stays in line with the `XProvider` naming convention? I find it a lot easier to read that way